src/Extensions/Hosting.Services/HostBuilderExtensions.cs

```diff
@@ -25,57 +26,53 @@ public static class HostBuilderExtensions
 		/// <summary>
 		/// Add actions that will be executed inside service fabric service RunAsync method
 		/// </summary>
-		public static ServiceFabricHostBuilder<TContext> AddServiceAction<TContext>(
-			this ServiceFabricHostBuilder<TContext> builder,
-			Func<TContext, CancellationToken, Task> action)
-				where TContext : ServiceContext =>
-					builder.AddServiceAction(new ServiceAction<TContext>(action));
+		public static ServiceFabricHostBuilder<TService> AddServiceAction<TService>(
+			this ServiceFabricHostBuilder<TService> builder,
```

Comment:
>builder [](start = 43, length = 7)

I'd argue that if you're going to add a `<summary>`, you should also add `<returns>`, `<param>` and `<typeparam>` comments although I realise you feel differently about this. If you're making this change, it applies throughout

Reply:
I'll add type parameters to explain limitations